extensions/amp-story/1.0/bookend/components/article.js

```diff
@@ -35,27 +33,24 @@ export let BookendArticleComponentDef;
 
 /**
  * Builder class for the small article component.
+ * @implements {BookendComponentInterface}
  */
-export class ArticleComponent extends AbstractBookendComponent {
-
+export class ArticleComponent {
   /** @override */
-  static isValid(articleJson) {
+  isValid(articleJson) {
     if (!articleJson['title'] || !articleJson['url']) {
-      user().error(TAG,
-          'Articles must contain `title` and `url` fields, skipping invalid.');
-      return false;
+      return user().assert(false, 'Articles must contain `title` and `url` ' +
```

Comment:
Sorry, I'm confused by the `user().assert(false, ...)` paradigm here.  Isn't this functionally different from the `user.error(...); return false;` paradigm used in the old version?

I would have expected the old version to report an error when this is invalid, then return false.
I would expect the new version to report this error and throw an exception when this is invalid.

I think we should pick which paradigm is the desired one, but if it's this, then we should change the `isValid` name to something like `assertValidity` and remove its return value.